### DISCUSSION TITLE
kitty command never returns on macOS

### DISCUSSION THREAD

**Discussion by mil-ad** [kovidgoyal/kitty#7979] [Q&A] [answered]
I'm trying to spawn a new kitty instance and run something from an existing kitty terminal. e.g:

```
kitty -- sh --noprofile --norc -c 'sleep 5'
```

I can see the new terminal window, it runs the command and the window disappears, however the `kitty` command hangs and never returns. This doesn't happen on Linux but happens on macOS.

**Comment by kovidgoyal** [comment, marked as answer]
https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.macos_quit_when_last_window_closed

On Thu, Oct 17, 2024 at 08:39:50AM -0700, Milad Alizadeh wrote:
> I'm trying to spawn a new kitty instance and run something from an existing kitty terminal. e.g:
>
> ```
> kitty -- sh --noprofile --norc -c 'sleep 5'
> ```
>
> I can see the new terminal window, it runs the command and the window disappears, however the `kitty` command hangs and never returns. This doesn't happen on Linux but happens on macOS.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/7979
> You are receiving this because you are subscribed to this thread.
>
> Message ID: ***@***.***>

--
_____________________________________

Dr. Kovid Goyal
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Comment by mil-ad** [comment]
Nice! 

I saw a reference to `open` on the man page and that seems to work too:

```
open -a kitty.app -n --args sh --noprofile --norc -c 'sleep 5'
```
